src/app/GitExtUtils/GitUI/ControlDpiExtensions.cs

```diff
@@ -102,6 +103,31 @@ public static void AdjustForDpiScaling(this Control control)
                         upDown.Margin = DpiUtil.Scale(new Padding(3));
                         break;
                     }
+
+                    case ToolStrip toolStrip:
+                    {
+                        foreach (ToolStripItem toolStripItem in toolStrip.FindToolStripItems())
+                        {
+                            switch (toolStripItem)
+                            {
+                                case ToolStripComboBox toolStripComboBox:
+                                {
+                                    if (toolStripComboBox.Tag as string != "__DPI_SCALED__")
```

Comment:
This _could_ lead to issues if `Tag` is set to something...
I suggest to add `DebugHelpers.Fail` check to alert developers in event the tag is set explicitly.

Reply:
Good catch... 🤦 don't really know yet how to solve it. Suggestions?

![image](https://github.com/user-attachments/assets/d276359e-d0f8-44ea-bc55-beddfc7202d7)

Reply:
yes or https://github.com/gitextensions/gitextensions/pull/11865#discussion_r1749255459 with your message

Reply:
@mstv I don't see how this solves the problem. There are two separate use cases that want to use a `Tag` property and neither would work if the other one took over. After some googling I've seen people use a `Dictionary<string, object>` and some wrapper methods to add support for multiple tags on a control.

Reply:
@mdonatas controls should be scaled in the .ctor, which means the scaling should happen once. Am I missing something?

Reply:
You are probably not missing anything... I've just seen other cases where this tag was used and assumed that was necessary and then fell into the rabbit hole :)
Removed the check. Scaling still works as expected.